src/ui/prompts/variablePrompts.ts

```diff
@@ -50,20 +55,30 @@ export const variableTypePrompt = {
     choices: CreateVariableDto.shape.type.options
 }
 
-export const variableValueStringPrompt = (variableKey: string, defaultValue?: string): Prompt => {
+export const variableValueStringPrompt = (
+    entity: VariablePromptEntity, 
+    defaultValue?: string
+): Prompt => {
+    const key = entity.value.key
+    const messageIdentifier = entity.type === 'variation' ? entity.value.name : entity.value.key
     return {
-        name: variableKey,
+        name: key,
         default: defaultValue,
         type: 'input',
-        message: `Variable value for ${variableKey}`,
+        message: `Variable value for ${messageIdentifier}`,
```

Comment:
it seems like this is duplicated for each type of prompt? we could create a utility method for building the base prompt and then add any additional properties specific to each type

Reply:
yup agreed